### PR TITLE
fix(fixRequestBody): fix request body for empty JSON object requests

### PR DESCRIPTION
**Pull request by mhassan1**
## Description

This PR fixes the `fixRequestBody` utility so that it treats empty object bodies like non-empty object bodies and fixes them, as expected.

## Motivation and Context

The `fixRequestBody` utility contains a check for empty objects that results in early `return`: https://github.com/chimurai/http-proxy-middleware/blob/c9bec81e82664809c816250f15d30543c4447571/src/handlers/fix-request-body.ts#L11-L13

Currently, JSON requests with empty object payloads do not get their request bodies fixed by this utility, and the request to the target hangs while the target waits for a payload that will never arrive.

This PR removes the special handling for empty objects and treats them like any other object.

Resolves https://github.com/chimurai/http-proxy-middleware/issues/639.

## How has this been tested?

I ran the reproduction steps in https://github.com/chimurai/http-proxy-middleware/issues/639 before and after the code change. I also updated automated tests.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by HappyColour** [comment]
This PR has been released right?

**Comment by eemelipa** [comment]
> This PR has been released right?

I don't think so. The PR is still open and the library doesn't have the fix

**Comment by HappyColour** [comment]
> > This PR has been released right?
> 
> I don't think so. The PR is still open and the library doesn't have the fix

Thx bro @eemelipa,please confirm this pr has been merged @mhassan1.

**Comment by mhassan1** [comment]
No, it is still open.

**Comment by chimurai** [comment]
Hi @mhassan1. I left a remark in the PR a while ago which hasn't been resolved.

Happy to merge it after it's fixed.

**Review by chimurai — CHANGES_REQUESTED** [review]

File: test/unit/fix-request-body.spec.ts [hunk 1]
```diff
@@ -24,19 +24,7 @@ describe('fixRequestBody', () => {
     expect(proxyRequest.write).not.toHaveBeenCalled();
   });
 
-  it('should not write when body is empty', () => {
```
[the remark this answers is not in the document]
Comment (mhassan1) [reply test/unit/fix-request-body.spec.ts:27]: Thanks, I've incorporated that feedback.

**Comment by mhassan1** [comment]
@chimurai This is ready for re-review.

**Review by chimurai — APPROVED** [review]

**Comment by chimurai** [comment]
Thanks @mhassan1 !

**Comment by chimurai** [comment]
published in `http-proxy-middleware@2.0.2`

**Pull request merged**